src/lib.rs

```diff
@@ -433,12 +433,20 @@ impl<'input, Endian> Unit<'input, Endian>
 
             // Extract source file and line information about the compilation unit
             let line_offset = match entry.attr_value(gimli::DW_AT_stmt_list) {
-                Some(gimli::AttributeValue::DebugLineRef(offset)) => offset,
+                Ok(Some(gimli::AttributeValue::DebugLineRef(offset))) => offset,
+                Err(e) => {
+                    return Err(Error::from(ErrorKind::Gimli(e)))
+                        .chain_err(|| "invalid compilation unit statement list")
+                }
                 _ => return Ok(None),
             };
             let comp_dir = entry.attr(gimli::DW_AT_comp_dir)
+                .map_err(|e| Error::from(ErrorKind::Gimli(e)))
```

Comment:
Ah, something different is actually going on from what I thought. The former is probably the error format we want. The latter just happens to also be a valid `Result` due to the `Error` wrapping with the `Msg`. The difference between the two is that the former also includes information about where the error came from (namely, that it came from `gimli`), which lets users of the API get a better sense of the true origin of errors. If you look at the defined `error_chain!` towards the top of `src/lib.rs`, you'll see that there's a separate `ErrorKind` used to wrap `gimli` errors, and I think we should make use of it. The reason they both work is that `next_error` is a `Option<Box<error::Error>>`, but `error-chain` encourages you to explicitly type all foreign errors to preserve their origin chain.